lib/public/components/Filters/common/filters/TextTokensFilterModel.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+import { Observable } from '/js/src/index.js';
+
+/**
+ * Model for text input
+ */
+export class TextTokensFilterModel extends Observable {
+    /**
+     * Constructor
+     * @param {RegExp} [inputValidationPattern] regex to validate input
+     * @param {RegExp|string} [splitPattern = null] regex for splitting input into tokens,
+     * it should describe delimiter between tokens. It might be single comma e.g. ',' or RegEx like /[ ,;]+/
+     * By defauly it's null so input is not splitted
+     */
+    constructor(inputValidationPattern, splitPattern = null) {
+        super();
+        this._inputValidationPattern = inputValidationPattern;
+        this._splitPattern = splitPattern;
+        this._raw = '';
+        this._visualChange$ = new Observable();
+    }
+
+    /**
+     * Update value kept by a filter model and inform observers that some change occured
+     * @param {string} value value to be stored
+     * @return {void}
+     */
+    update(value) {
+        const { _raw: previousRaw } = this;
+        this._raw = value;
+        if (this._normalizedEqual(value) || previousRaw === value) {
+            if (this.isValid) {
+                this.notify();
+            } else {
+                this._visualChange$.notify();
+            }
+        } else {
+            this.notify();
+        }
+    }
+
+    /**
+     * Getter to raw value stored by a model
```

Comment:
by a model? What does that means?
`Returns the raw value of the filter`

Reply:
refactored